core/src/query.rs

```diff
@@ -59,90 +60,134 @@ impl MongoQuery {
-                    "collectionName": "$_id",
-                    "schema": "$schema"
+                },
+                doc! {"$project":{
+                    "_id": 1,
+                    "schema": 1
                     }
-                }
-            },
-            doc! {"$project": {
-                "_id": 0,
-                current_db: {
-                    "$arrayToObject": [{
-                        "$map": {
-                            "input": "$collections",
-                            "as": "coll",
-                            "in": {
-                                "k": "$$coll.collectionName",
-                                "v": "$$coll.schema"
-                                }
+                },
+                doc! {"$group": {
+                    "_id": null,
+                    "collections": {
+                        "$push": {
+                            "collectionName": "$_id",
+                            "schema": "$schema"
                             }
-                        }]
+                        }
                     }
+                },
+                doc! {"$project": {
+                    "_id": 0,
+                    current_db: {
+                        "$arrayToObject": [{
+                            "$map": {
+                                "input": "$collections",
+                                "as": "coll",
+                                "in": {
+                                    "k": "$$coll.collectionName",
+                                    "v": "$$coll.schema"
+                                    }
+                                }
+                            }]
+                        }
+                    }
+                },
+            ];
+
+            // create the schema_catalog document
+            let mut schema_catalog_doc_vec: Vec<Document> = client.runtime.block_on(async {
+                schema_collection
+                    .aggregate(schema_catalog_aggregation_pipeline)
+                    .await
+                    .map_err(Error::QueryExecutionFailed)?
+                    .try_collect::<Vec<Document>>()
+                    .await
+                    .map_err(Error::QueryExecutionFailed)
+            })?;
+
+            if schema_catalog_doc_vec.len() > 1 {
+                return Err(Error::MultipleSchemaDocumentsReturned(
+                    schema_catalog_doc_vec.len(),
+                ));
+            } else if schema_catalog_doc_vec.is_empty() {
+                log::warn!("There is a `__sql_schemas` collection in database `{0}`; however, no schema information was returned for the requested collections.\
+                This means that all the requested collections either don't exists in `{0}` or don't have an entry in the `__sql_schemas` collection. For now, they will\
+                be assigned empty schemas. Hint: You either need to generate schemas for your collections or correct your query.", current_db);
```

Comment:
Use a similar warning message to the one below. I would like to avoid putting too much technical details in the message.

```suggestion
                log::warn!("No schema information found for the collections {0} in database `{1}`.\
                Either the collections don't exists in `{1}` or they don't have a schema. Hint: You either need to generate schema for your collections or correct your query.", collection_names, current_db);
```